packages/flutter_tools/lib/src/android/gradle.dart

```diff
@@ -202,466 +203,608 @@ void createSettingsAarGradle(Directory androidDirectory) {
   globals.printStatus('$successMark `$newSettingsRelativeFile` created successfully.');
 }
 
-/// Builds an app.
-///
-/// * [project] is typically [FlutterProject.current()].
-/// * [androidBuildInfo] is the build configuration.
-/// * [target] is the target dart entry point. Typically, `lib/main.dart`.
-/// * If [isBuildingBundle] is `true`, then the output artifact is an `*.aab`,
-///   otherwise the output artifact is an `*.apk`.
-/// * The plugins are built as AARs if [shouldBuildPluginAsAar] is `true`. This isn't set by default
-///   because it makes the build slower proportional to the number of plugins.
-/// * [retries] is the max number of build retries in case one of the [GradleHandledError] handler
-///   returns [GradleBuildStatus.retry] or [GradleBuildStatus.retryWithAarPlugins].
-Future<void> buildGradleApp({
-  @required FlutterProject project,
-  @required AndroidBuildInfo androidBuildInfo,
-  @required String target,
-  @required bool isBuildingBundle,
-  @required List<GradleHandledError> localGradleErrors,
-  bool shouldBuildPluginAsAar = false,
-  int retries = 1,
-}) async {
-  assert(project != null);
-  assert(androidBuildInfo != null);
-  assert(target != null);
-  assert(isBuildingBundle != null);
-  assert(localGradleErrors != null);
-  assert(globals.androidSdk != null);
-
-  if (!project.android.isUsingGradle) {
-    _exitWithProjectNotUsingGradleMessage();
-  }
-  if (!_isSupportedVersion(project.android)) {
-    _exitWithUnsupportedProjectMessage();
+/// An implementation of the [AndroidBuilder] that delegates to gradle.
+class AndroidGradleBuilder implements AndroidBuilder {
```

Comment:
The only change here is that I made this a class instead of multiple top level methods.